synapse/storage/engines/sqlite.py

```diff
@@ -12,14 +12,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import sqlite3
 import struct
 import threading
+import typing
 
 from synapse.storage.engines import BaseDatabaseEngine
 
+if typing.TYPE_CHECKING:
```

Comment:
wait, I don't think this exists in python 3.5.1? (which rather begs the question of how the CI is working).

```suggestion
MYPY = False
if MYPY:
```

Reply:
sigh. It turns out that the package called "python3 3.5.1-3" in ubuntu xenial is actually python 3.5.2. So stuff that only exists in 3.5.2 is fine

Reply:
This got me curious, so I grepped synapse for the two styles. It looks like the `MYPY` "constant" is used in 5 places (4 of which already import typing), but `typing.TYPE_CHECKING` is already used in `synapse/logging/context.py` as well. It'd probably be worth standardising on `TYPE_CHECKING` at some point.

Reply:
yes it would. I think the MYPY thing was done in the mistaken belief that we needed to support python 3.5.1.